src/shared/firewall-util-nft.c

```diff
@@ -812,6 +839,66 @@ static int fw_nftables_recreate_table(sd_netlink *nfnl, int af, sd_netlink_messa
         return 0;
 }
 
+static int nft_message_add_setelem_ip6range(sd_netlink_message *m,
+                                            const union in_addr_union *source,
+                                            unsigned int prefixlen) {
+        struct in6_addr start, end;
+        uint8_t carry = 0;
+        int r, i, j;
+
+        assert(prefixlen <= 128);
+
+        for (i = 0, j = 15; i < 16; i++) {
+                uint8_t nm;
+
+                nm = 0xFF;
+                if (prefixlen < 8)
+                        nm = 0xFF << (8 - prefixlen);
+
+                start.s6_addr[i] = source->in6.s6_addr[i] & nm;
+                end.s6_addr[i] = start.s6_addr[i];
+                if (prefixlen <= 8 && j == 15) {
+                         carry = 1u << (8 - prefixlen);
```

Comment:
If `prefixlen == 0`, then carry = 0, is it expected?

Reply:
prefixlen == 0 should not occur within the branch, the caller has a check for prefixlen == 0 and errors out.
I can add an assertion if you like.